### PR TITLE
Chore: Dockerizing Setup

### PR DESCRIPTION
**Pull request by HemlockPham7**
- [x] Set up config to connect database
- [x] Connect to database
- [x] Get data from db instead of mock data
- [x] Set up docker
- [x] Dockerize successfully

### CODE REVIEW

**Comment by ndkhoa-harry** [comment]
> [x] Set up config to connect database [x] Connect to database [x] Get data from db instead of mock data [x] Set up docker [ ] Dockerize successfully

Wrong markdown format, please check

**Comment by ndkhoa-harry** [comment]
the name of this branch should be `chore/dockerizing-setup`

**Comment by ndkhoa-harry** [comment]
Please minimize your commits as much as possible, each commit should describe a simple task, don't just commit all files at once.

File: backend/cmd/api/main.go [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
 	"github.com/gofiber/fiber/v2"
 )
 
 func main() {
+	envConfig := config.NewEnvConfig()
+	db := db.Init(envConfig, db.DBMigrator)
+
 	app := fiber.New(fiber.Config{
 		AppName:      "TicketBooking",
 		ServerHeader: "Fiber",
```
Comment (ndkhoa-harry) [inline backend/cmd/api/main.go]: Should put these in `.env` file

File: backend/db/database.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/HemlockPham7/backend/config"
+	"github.com/gofiber/fiber/v2/log"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+)
+
+func Init(config *config.EnvConfig, DBMigrator func(*gorm.DB) error) *gorm.DB {
+	uri := fmt.Sprintf(`
+		host=%s user=%s dbname=%s password=%s sslmode=%s port=5432`,
```
Comment (ndkhoa-harry) [inline backend/db/database.go]: Should put `port` to env file, too

File: backend/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FROM golang:1.22.2-alpine3.19
+
+WORKDIR /src/app
+
+RUN go install github.com/cosmtrek/air@latest
```
Comment (ndkhoa-harry) [inline backend/Dockerfile]: Why'd you install this library in docker?

Comment (HemlockPham7) [reply backend/Dockerfile]: This setup helps streamline the development process by automatically rebuilding and restarting the server whenever I make code changes, which can be very useful for testing and debugging without having to restart the container manually.

File: backend/db/database.go (hunk 2 above)
Comment (HemlockPham7) [reply backend/db/database.go]: Resolved, please recheck.

File: backend/cmd/api/main.go (hunk 1 above)
Comment (HemlockPham7) [reply backend/cmd/api/main.go]: Resolved, please recheck.

**Comment by ndkhoa-harry** [comment]
The message of last commit should be `fix: comments`

**Review by ndkhoa-harry — APPROVED** [review]
LGTM 💯

**Pull request merged**